app/uk/gov/hmrc/gform/formtemplate/FormComponentMaker.scala

```diff
@@ -157,7 +157,11 @@ class FormComponentMaker(json: JsValue) {
       case JsUndefined() => Right(None)
     }
 
-  lazy val dividerPositon: Option[Int] = (json \ "dividerPosition").asOpt[Int]
+  lazy val dividerPositon: Option[DividerPosition] =
+    for {
+      s <- (json \ "dividerPosition").asOpt[String]
+    } yield Try(s.toInt).toOption.fold[DividerPosition](DividerPosition.Value(s))(i => DividerPosition.Number(i))
```

Comment:
Your comment [here](https://github.com/hmrc/gform/pull/918#discussion_r1142011679) pique my interest. 
Int and String are actually easily distinguishable, and we should not do `Try(s.toInt)`

Mapping:
```
"dividerPosition": 4, // -> DividerPosition.Number
"dividerPosition": "4", // -> DividerPosition.Value
```
Given this mapping we can match on json types without need for `Try`

Reply:
I refactored to remove Try